hudi-client/hudi-spark-client/src/main/scala/org/apache/spark/sql/HoodieInternalRowUtils.scala

```diff
@@ -210,6 +220,26 @@ object HoodieInternalRowUtils {
     }
   }
 
+  def getCachedAvroSerializer(schema: Schema): HoodieAvroSerializer = {
+    val avroSerializerMap = avroSerializerMapThreadLocal.get()
+    if (!avroSerializerMap.containsKey(schema)) {
```

Comment:
Lookups w/ Avro schema as keys are costly -- let's minimize number of lookups we need to make:

 - We can use `getOrElseUpdate` to get/update in one lookup
 - We can store pair (StructType, SerDe) to avoid the need to do 2 lookups